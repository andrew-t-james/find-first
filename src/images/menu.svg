<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px"
fill="#EAECEE" viewBox="0 0 396.9 496.125" enable-background="new 0 0 396.9 396.9" xml:space="preserve"><g><rect x="85" y="258.4" width="226.8" height="14.6"/></g><g><rect x="85" y="124" width="226.8" height="14.6"/></g><g><rect x="85" y="191.2" width="138.7" height="14.6"/></g></svg>