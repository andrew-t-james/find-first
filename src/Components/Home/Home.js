import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SlideMenu from 'react-slide-menu';
import { toggleMenu } from '../../Actions/menu';
import { nav } from '../../helpers/nav';

import SignIn from '../SignIn/SignIn';

import menuIcon from '../../images/menu.svg';

const logoComponent = () => (
  <img src={logo} />
);

export const Home = props => {
  const { slideMenuActive, toggleMenu } = props;

  return (
    <SlideMenu
      active={slideMenuActive}
      nav={nav}
      reactRouter={true}
      closeMenu={() => toggleMenu()}
      extraComponent={logoComponent}
    >
      <section className="home">
        <div className={`home__menu--icon wrapper-menu ${slideMenuActive ? 'open' : ''}`} onClick={toggleMenu}>
          <div className="line-menu half start"></div>
          <div className="line-menu"></div>
          <div className="line-menu half end"></div>
        </div>
        <div className="logo-section">
          <div className="logo">
            <h1 className="logo__forward">F</h1>
            <h1 className="logo__backward">F</h1>
          </div>
          <div className="logo__title">
            <h1>Find<span className="logo__title-thin">First</span></h1>
          </div>
        </div>
      </section>
    </SlideMenu>
  );
};

export const mapStateToProps = state => ({
  slideMenuActive: state.slideMenuActive
});

export const mapDispatchToProps = dispatch => ({
  toggleMenu: () => dispatch(toggleMenu())
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);

Home.propTypes = {
  toggleMenu: PropTypes.func,
  slideMenuActive: PropTypes.bool
};