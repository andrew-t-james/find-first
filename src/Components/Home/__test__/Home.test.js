import React from 'react';
import { shallow } from 'enzyme';
import { Home, mapStateToProps, mapDispatchToProps } from '../Home';
import { toggleMenu } from '../../../Actions/menu';

jest.mock('../../../firebase/firebase.js');

describe('<Home />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Home toggleMenu={jest.fn()} />));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    test('should map slideMenuActive to state', () => {
      const mockState = {
        slideMenuActive: true
      };

      const expectedState = {
        slideMenuActive: true
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch when toggleMenu is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleMenu();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.toggleMenu();
      expect(mockDispatch).toHaveBeenCalled();
    });

  });

});
