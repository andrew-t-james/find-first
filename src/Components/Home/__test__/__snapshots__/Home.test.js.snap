// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<e
  active={true}
  closeMenu={[Function]}
  nav={
    Array [
      Object {
        "id": "home",
        "label": "Home",
        "path": "/",
      },
      Object {
        "id": "sign-in",
        "label": "Sign-In",
        "path": "/sign-in",
      },
      Object {
        "id": "dashboard",
        "label": "Dashboard",
        "path": "/dashboard",
      },
      Object {
        "id": "saved-jobs",
        "label": "Saved Jobs",
        "path": "/saved-jobs",
      },
    ]
  }
  reactRouter={true}
>
  <section
    className="home"
  >
    <div
      className="home__menu--icon wrapper-menu open"
      onClick={[MockFunction]}
    >
      <div
        className="line-menu half start"
      />
      <div
        className="line-menu"
      />
      <div
        className="line-menu half end"
      />
    </div>
    <div
      className="logo-section"
    >
      <div
        className="logo"
      >
        <h1
          className="logo__forward"
        >
          F
        </h1>
        <h1
          className="logo__backward"
        >
          F
        </h1>
      </div>
      <div
        className="logo__title"
      >
        <h1>
          Find
          <span
            className="logo__title-thin"
          >
            First
          </span>
        </h1>
      </div>
    </div>
  </section>
</e>
`;
